pkg/controller/persistentvolume/persistentvolume_claim_binder_controller.go

```diff
@@ -166,96 +189,87 @@ func syncVolume(volumeIndex *persistentVolumeOrderedIndex, binderClient binderCl
 			// to start the volume again at the beginning of this lifecycle.
 			// ClaimRef is the last bind between persistent volume and claim.
 			// The claim has already been deleted by the user at this point
-			oldClaimRef := volume.Spec.ClaimRef
-			volume.Spec.ClaimRef = nil
-			_, err = binderClient.UpdatePersistentVolume(volume)
+			newVolume, err := copyVolume(volume)
 			if err != nil {
-				// rollback on error, keep the ClaimRef until we can successfully update the volume
-				volume.Spec.ClaimRef = oldClaimRef
+				return err
+			}
+			newVolume.Spec.ClaimRef = nil
+			if _, err = binderClient.UpdatePersistentVolume(newVolume); err != nil {
```

Comment:
Instead of _, wouldn't we need the returned PV?  It would have new updated timestamps, resource versions,etc from the API server.